integration-tests/src/test_loop/tests/state_sync.rs

```diff
@@ -356,11 +380,40 @@ fn slow_test_state_sync_current_epoch() {
                 .iter()
                 .map(|(shard_id, produced)| (*shard_id, produced.to_vec()))
                 .collect(),
+            t.skip_sync_block,
         );
         run_test(state);
     }
 }
 
+// Here we drop the block that's supposed to be the sync hash after the first full epoch,
+// which causes it to be produced but then skipped on the final chain. If the state sync code
+// is unaware of the possibility of forks, this will cause the producer of that block to
+// believe that that block should be the sync hash block, while all other nodes will
+// believe it should be the next block.
```

Comment:
Can you also assert that the fork actually happens?

Reply:
added an `assert_fork_happened()` function